storage/src/storage.rs

```diff
@@ -21,9 +21,23 @@ pub trait Storage {
     /// Delete a value from the storage
     fn delete(&self, key: &[u8]) -> Result<()>;
 
-    /// Create an iterator over all the keys that start with the given prefix
+    /// Create an iterator over all the keys that start with the given prefix, in undefined order
     fn prefix_iterator<'a, 'b: 'a>(&'a self, prefix: &'b [u8]) -> Result<StorageIterator<'a>>;
 
+    /// Create an iterator over all the keys that start with the given prefix, in order, starting
+    /// from the first key.
+    fn prefix_iterator_forward<'a, 'b: 'a>(
+        &'a self,
+        prefix: &'b [u8],
+    ) -> Result<StorageIterator<'a>>;
+
+    /// Create an iterator over all the keys that start with the given prefix, in reverse, starting
+    /// from the last key.
+    fn prefix_iterator_reverse<'a, 'b: 'a>(
+        &'a self,
+        prefix: &'b [u8],
+    ) -> Result<StorageIterator<'a>>;
+
```

Comment:
Actually, it may be better to only have a single method in this API, something like `iterate_range(start: &[u8], end: &[u8]) -> DoubleEndedIterator`